### ISSUE TITLE
__slots__ are ignored

### ISSUE THREAD

**Issue by zupo** [Fatal1ty/mashumaro#47]
Hey there! First, thanks for this great library, I am really enjoying using it!

Sometimes I like to use `__slots__` with dataclasses to prevent me from accidentally setting a value to a (typo-ed) new attribute instead of to an existing attribute. 

However, if I use the `DataClassDictMixin` from mashumaro, it seems like `__slots__` are ignored. I.e. If I set a value to a non-existing attribute, I do not get AttributeError, but a new attribute is created and value is set to it, as if `__slots__` would not exist.

Here's a short pytest case to showcase the problem:

```python
import pytest
from dataclasses import dataclass
from mashumaro import DataClassDictMixin


@dataclass
class Foo:
    __slots__ = ["number"]
    number: int


@dataclass
class Bar(DataClassDictMixin):
    __slots__ = ["number"]
    number: int


def test_slots():

    foo = Foo(1)
    with pytest.raises(AttributeError) as err:
        foo.new_attribute = 2
    assert str(err.value) == "'Foo' object has no attribute 'new_attribute'"

    bar = Bar(1)
    bar.new_attribute = 2
    # -> should also fail with "'Bar' object has no attribute 'new_attribute'",
    # but it doesn't
```

**Comment by Fatal1ty** [comment]
Hello @zupo. Thank you for creating this issue.

I came up with the idea of adding empty `__slots__` to the parent classes to fix this issue. It seems to work and helps to reduce memory consumption. You can check this branch https://github.com/Fatal1ty/mashumaro/tree/slots.

**Comment by zupo** [comment]
Hmm, I tried, but it doesn't seem to work?

**Comment by Fatal1ty** [comment]
> Hmm, I tried, but it doesn't seem to work?

I checked your example in the new branch and it fails as it should with `AttributeError: 'Bar' object has no attribute 'new_attribute'`. The slots test also [passes](https://github.com/Fatal1ty/mashumaro/actions/runs/728963350) on different Python versions. Can you shed some light on what's not working?

**Comment by Fatal1ty** [comment]
> Hmm, I tried, but it doesn't seem to work?

I checked your example in the new branch and it fails as it should with `AttributeError: 'Bar' object has no attribute 'new_attribute'`. The [slots test](https://github.com/Fatal1ty/mashumaro/blob/slots/tests/test_common.py#L13) also [passes](https://github.com/Fatal1ty/mashumaro/actions/runs/728963350) on different Python versions. Can you shed some light on what's not working?

**Comment by zupo** [comment]
Gosh, I'm terribly sorry for the noise, it was a mistake on my end. Works as expected!

**Comment by zupo** [comment]
https://github.com/Fatal1ty/mashumaro/commit/543356e4add2e215f90d36e011efc489e0d2e936 is definitely needed.

I was not using the correct branch when I first tested (I thought I did) and now I re-checked and works as expected with the `slots` branch.

I closed the issue by mistake. Getting late, I should go to bed.

**Comment by Fatal1ty** [comment]
Fixed in 2.2.
